modules/code-deployment/src/main/java/org/apache/ignite/internal/deployunit/DeploymentManagerImpl.java

```diff
@@ -0,0 +1,424 @@
+                        if (builder != null) {
+                            result.complete(builder.build());
+                        } else {
+                            result.completeExceptionally(
+                                    new DeploymentUnitNotExistException("Unit with " + id + "not exist."));
+                        }
+                    }
+                });
+        return result;
+    }
+
+    @Override
+    public void start() {
+        unitsFolder = unitsFolder.resolve(configuration.deploymentLocation().value());
+        clusterService.messagingService().addMessageHandler(DeployUnitMessageTypes.class,
+                (message, senderConsistentId, correlationId) -> {
+                    if (message instanceof DeployUnitRequest) {
+                        processDeployRequest((DeployUnitRequest) message, senderConsistentId, correlationId);
+                    } else if (message instanceof UndeployUnitRequest) {
+                        processUndeployRequest((UndeployUnitRequest) message, senderConsistentId, correlationId);
+                    }
+                });
+    }
+
+    private void processDeployRequest(DeployUnitRequest executeRequest, String senderConsistentId, long correlationId) {
+        doDeploy(executeRequest).whenComplete((success, throwable) -> {
+            DeployUnitResponseBuilder builder = DeployUnitResponseImpl.builder();
+            if (throwable != null) {
+                builder.error(throwable);
+            }
+            clusterService.messagingService().respond(senderConsistentId,
+                    builder.build(), correlationId);
+        });
+    }
+
+    private void processUndeployRequest(UndeployUnitRequest executeRequest, String senderConsistentId, long correlationId) {
+        try {
+            Path unitPath = unitsFolder
+                    .resolve(executeRequest.id())
+                    .resolve(executeRequest.version());
+
+            IgniteUtils.deleteIfExistsThrowable(unitPath);
+        } catch (IOException e) {
+            LOG.error("Failed to undeploy unit " + executeRequest.id() + ":" + executeRequest.version(), e);
+            clusterService.messagingService()
+                    .respond(senderConsistentId, UndeployUnitResponseImpl.builder().error(e).build(), correlationId);
+            return;
+        }
+
+        clusterService.messagingService()
+                .respond(senderConsistentId, UndeployUnitResponseImpl.builder().build(), correlationId);
+    }
+
+    private CompletableFuture<Boolean> doDeploy(DeployUnitRequest executeRequest) {
+        String id = executeRequest.id();
+        String version = executeRequest.version();
+        try {
+            Path unitPath = unitsFolder
+                    .resolve(executeRequest.id())
+                    .resolve(executeRequest.version())
+                    .resolve(executeRequest.unitName());
+            Path unitPathTmp = unitPath.resolveSibling(unitPath.getFileName() + ".tmp");
```

Comment:
I think strings like ".tmp" should be in constants. Same goes for separators, like ":", that you were using earlier